libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.NetFx/BotFrameworkConfigurationBuilder.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using Microsoft.Bot.Builder.Middleware;
+
+namespace Microsoft.Bot.Builder.Integration.AspNet
+{
+    public class BotFrameworkConfigurationBuilder
+    {
+        private BotFrameworkOptions _options;
+
+        public BotFrameworkConfigurationBuilder()
+        {
+            _options = new BotFrameworkOptions();
+        }
+
+        public BotFrameworkOptions BotFrameworkOptions { get => _options; }
+
+        public BotFrameworkConfigurationBuilder UseApplicationIdentity(string applicationId, string applicationPassword)
+        {
+            _options.AppId = applicationId;
+            _options.AppPassword = applicationPassword;
+
+            return this;
+        }
```

Comment:
Can this use ICredentialProvider instead. ICredential provider is reusable and in case a middleware wants to get it's own tokens (for example MsTeams extension) dev should have to pass same info over and over.

Reply:
Great point! We're trying to deliver this PR now, but I have added #237 to track and add it in a follow up PR.